File: java/com/epicdragonworld/gameserver/network/packets/receivable/CharacterSelectionInfoRequest.java
package com.epicdragonworld.gameserver.network.packets.receivable;

import com.epicdragonworld.gameserver.network.GameClient;
import com.epicdragonworld.gameserver.network.ReceivablePacket;
import com.epicdragonworld.gameserver.network.packets.sendable.CharacterSelectionInfoResult;

/**
 * @author Pantelis Andrianakis
 */
public class CharacterSelectionInfoRequest
{
	public CharacterSelectionInfoRequest(GameClient client, ReceivablePacket packet)
	{
		// Read data.
		final String accountName = packet.readString().toLowerCase();
		
		// If account has logged send the information.
		if (client.getAccountName().equals(accountName))
		{
			client.channelSend(new CharacterSelectionInfoResult(accountName));
		}
	}
}

File: EpicDragonWorldServer/Network/Packets/Receivable/CharacterSelectionInfoRequest.cs
/**
 * Author: Pantelis Andrianakis
 * Date: November 7th 2018
 */
public class CharacterSelectionInfoRequest
{
    public CharacterSelectionInfoRequest(GameClient client, ReceivablePacket packet)
    {
        // Read data.
        string accountName = packet.ReadString().ToLowerInvariant();

        // If account has logged send the information.
        if (client.GetAccountName().Equals(accountName))
        {
            client.ChannelSend(new CharacterSelectionInfoResult(accountName));
        }
    }
}
